src/exif.c

```diff
@@ -0,0 +1,119 @@
+                        return AVIF_RESULT_OK;
+                    case 2: // The 0th row is at the visual top of the image, and the 0th column is the visual right-hand side.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IMIR;
+                        image->irot.angle = 0; // ignored
+                        image->imir.mode = 1;
+                        return AVIF_RESULT_OK;
+                    case 3: // The 0th row is at the visual bottom of the image, and the 0th column is the visual right-hand side.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IROT;
+                        image->irot.angle = 2;
+                        image->imir.mode = 0; // ignored
+                        return AVIF_RESULT_OK;
+                    case 4: // The 0th row is at the visual bottom of the image, and the 0th column is the visual left-hand side.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IMIR;
+                        image->irot.angle = 0; // ignored
+                        image->imir.mode = 0;
+                        return AVIF_RESULT_OK;
+                    case 5: // The 0th row is the visual left-hand side of the image, and the 0th column is the visual top.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IROT | AVIF_TRANSFORM_IMIR;
+                        image->irot.angle = 1; // applied before imir according to MIAF spec ISO/IEC 28002-12:2021 - section 7.3.6.7
+                        image->imir.mode = 0;
+                        return AVIF_RESULT_OK;
+                    case 6: // The 0th row is the visual right-hand side of the image, and the 0th column is the visual top.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IROT;
+                        image->irot.angle = 3;
+                        image->imir.mode = 0; // ignored
+                        return AVIF_RESULT_OK;
+                    case 7: // The 0th row is the visual right-hand side of the image, and the 0th column is the visual bottom.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IROT | AVIF_TRANSFORM_IMIR;
+                        image->irot.angle = 3; // applied before imir according to MIAF spec ISO/IEC 28002-12:2021 - section 7.3.6.7
+                        image->imir.mode = 0;
+                        return AVIF_RESULT_OK;
+                    case 8: // The 0th row is the visual left-hand side of the image, and the 0th column is the visual bottom.
+                        image->transformFlags = otherFlags | AVIF_TRANSFORM_IROT;
+                        image->irot.angle = 1;
+                        image->imir.mode = 0; // ignored
+                        return AVIF_RESULT_OK;
+                    default: // reserved
+                        break;
+                }
+            }
+        }
+        if (!avifROStreamReadU32(&stream, &offsetToNextIfd)) {
```

Comment:
> IMPORTANT: Should we use `avifROStreamReadU16Endianness` here? If not, add a comment to assure the reader that this is correct.

Good catch.

> UPDATE: I didn't find any code in Chromium's third_party/blink/renderer/platform/image-decoders/jpeg/jpeg_image_decoder.cc that matches this exactly. The closest match is the `CheckExifHeader()` call, but that call will also read the TIFF header.

I followed JEITA CP-3451C rather than another project. Most implementations I saw on the web when searching for parsers were just lazily looking for the first Exif IFD orientation tag.